azerothcore/azerothcore-wotlk
# feat(Core/Scripting): Implement OnlyOnceAreaTrigger  & Zul'Gurub's in…

…tro speech



## Changes Proposed:
-  Implement OnlyOnceAreaTrigger scripts, to be used for areatriggers inside instances that should only fire once.
-  Scripted Zul'Gurub's entrance speech, so testers have anything to test it.

## Issues Addressed:

- Closes 

## SOURCE:


## Tests Performed:

- Tested in-game.
- 


## How to Test the Changes:


1. Enter Zul'Gurub and walk down the entrance stairway
2. Notice Hakkar will yell his thing
3. Walk back upstairs again, notice he doesn't repeat his text (onlyonceareatrigger)

## Known Issues and TODO List:




## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
